core/server/worker/src/main/java/alluxio/worker/block/BlockWorker.java

```diff
@@ -423,6 +424,28 @@ void closeUfsBlock(long sessionId, long blockId)
       throws BlockAlreadyExistsException, BlockDoesNotExistException, IOException,
       WorkerOutOfSpaceException;
 
+  /**
+   * Gets the block reader to read from Alluxio block or UFS block.
+   * This operation must be paired with {@link #cleanBlockReader(long, long, BlockReader)}.
+   *
+   * @param request the block read request
+   * @return a block reader to read data from
+   */
+  BlockReader getBlockReader(BlockReadRequest request) throws IOException,
```

Comment:
I would suggest to pass `long sessionId, long blockId` instead of `BlockReadRequest`, to keep `getBlockReader` and `cleanBlockReader` in parity. Alternatively, pass `BlockReadRequest` to `cleanBlockReader`

Reply:
Agreed that we need to keep the two methods consistent.
The `BlockReadRequest` contains too much information, like isPromoted, isPersisted, UfsBlockOptions, etc. Putting them all in the constructor is kind of messy. I changed to pass `BlockReadRequest` to `cleanBlockReader`.